src/lambda-manager/template.yaml

```diff
@@ -85,157 +76,47 @@ Parameters:
-               - CoralogixRegionMap
-               - Ref: CoralogixRegion
-               - LogUrl
-        RequestConfiguration:
-          ContentEncoding: GZIP
-          CommonAttributes:
-            - AttributeName: "dynamicMetadata"
-              AttributeValue: "true"
-            - AttributeName: "integrationType"
-              AttributeValue: "CloudWatch_Lambda"
-        BufferingHints:
-          IntervalInSeconds: 60
-          SizeInMBs: 6
-        RetryOptions:
-          DurationInSeconds: 60
-        S3BackupMode: FailedDataOnly
-        S3Configuration:
-          BucketARN: !GetAtt 
-            - CoralogixFirehoseFailed
-            - Arn
-          CompressionFormat: UNCOMPRESSED
-          RoleARN: !GetAtt FirehoseLogsRole.Arn
-        CloudWatchLoggingOptions:
-          Enabled: true
-          LogGroupName: !Ref CoralogixFirehoseFailed
-          LogStreamName: !Ref CoralogixFirehoseFailed
-        ProcessingConfiguration:
-          Enabled: false
-        RoleARN: !GetAtt FirehoseLogsRole.Arn
-        
-  CloudwatchToFirehoseRole:
-    Type: AWS::IAM::Role
-    Properties:
-      AssumeRolePolicyDocument:
-        Version: '2012-10-17'
-        Statement:
-          - Sid: 'CloudwatchToFirehoseRole'
-            Effect: Allow
             Principal:
-              Service: !Join
-                - ''
-                - - 'logs.'
-                  - !Ref 'AWS::Region'
-                  - '.amazonaws.com'
+              Service: lambda.amazonaws.com
             Action: sts:AssumeRole
-  CloudwatchToFirehoseAttachePolicy:
-    Type: AWS::IAM::Policy
-    Properties:
-      PolicyName: "CloudWatchFirehosePermissionPolicy"
-      PolicyDocument:
-        Version: '2012-10-17'
-        Statement:
-          - Effect: Allow
-            Action:
-              - firehose:PutRecord
-              - firehose:PutRecordBatch
-              - firehose:UpdateDestination
-            Resource: !GetAtt CoralogixDeliveryStream.Arn
-      Roles:
-        - Ref: CloudwatchToFirehoseRole
+      Policies:
+        - PolicyName: LambdaExecutionPolicy
+          PolicyDocument:
+            Version: '2012-10-17'
+            Statement:
+              - Sid: CXLambdaUpdateConfig
+                Effect: Allow
+                Action:
+                  - lambda:UpdateFunctionConfiguration
+                Resource: !GetAtt LambdaFunction.Arn
+              - Sid: SNSPublishPermission
+                Effect: Allow
+                Action:
+                  - sns:Publish
+                Resource: "arn:aws:sns:us-east-1:771039649440:manager4-LambdaFunctionEventInvokeConfigOnFailureTopic-xB2dSpg5hshT"
   LambdaFunction:
     Type: AWS::Serverless::Function
     Properties:
       Description: Adds Coralogix firehose to Loggroups
       CodeUri: .
       Handler: lambda_function.lambda_handler
-      Runtime: python3.9
+      Runtime: python3.12
       Architectures:
-        - Ref: FunctionArchitecture
+        - arm64
```

Comment:
That won't work in some AWS regions as they don't have arm64 lambdas. x86_64 is a safer choice as it's present in all regions.